src/components/AddEvent.tsx

```diff
@@ -88,90 +81,103 @@ const AddEvent = () => {
     setWhenToRecur(e.currentTarget.id);
   };
 
-  const formSubmitHandler = (e: React.FormEvent) => {
-    e.preventDefault();
-    const randomId = Math.floor((1 + Math.random()) * 0x10000)
+  const newEvent: Event = {
+    id: Math.floor((1 + Math.random()) * 0x10000)
       .toString(16)
-      .substring(1);
-
-    const randomColor = `#${Math.floor(Math.random() * 16777215).toString(16)}`;
-    const value = (endTime - startTime) / 60;
-    const newEvent: Event = {
-      id: randomId,
-      title: eventTitle,
-      value: value,
-      color: randomColor,
-      start: startTime,
-      end: endTime,
-      recurrence,
-    };
+      .substring(1),
+    title: eventTitle,
+    value: (endTime - startTime) / 60,
+    color: `#${Math.floor(Math.random() * 16777215).toString(16)}`,
+    start: startTime,
+    end: endTime,
+    recurrence,
+  };
 
-    const getDates = (startDate: string, stopDate: string) => {
-      const dateArray = [];
-      let currentDate = new Date(startDate);
-      while (currentDate <= new Date(stopDate)) {
-        switch (whenToRecur) {
-          case "day":
-            dateArray.push(fromDateToString(currentDate));
-            currentDate = currentDate.addDays(1);
-            break;
-          case "workday":
-            if (currentDate.getDay() > 0 && currentDate.getDay() < 6) {
-              dateArray.push(fromDateToString(currentDate));
-            }
-            currentDate = currentDate.addDays(1);
-            break;
-          case "weekend":
-            if (currentDate.getDay() === 0 || currentDate.getDay() === 6) {
-              dateArray.push(fromDateToString(currentDate));
-            }
-            currentDate = currentDate.addDays(1);
-            break;
-        }
+  //confirmation handler for overlapping events
+  const confirmReplaceHandler = () => {
+    const lastDate = new Date(selectedDate).addDays(parseInt(recurrence));
+    const dates = createDatesArray(selectedDate, lastDate, whenToRecur);
+    let newState = [...data];
+    dates.forEach((date) => {
+      const existingDate = data.find((event) => event.date === date);
+      if (existingDate) {
+        existingDate.event.forEach((elem, i) => {
+          if (elem.start < newEvent.end && newEvent.start < elem.end) {
+            existingDate.event.splice(i, 1);
+          }
+        });
+        const index = (data as EventDate[]).indexOf(existingDate);
+        newState[index].event.push(newEvent);
+      } else {
+        newState.push({ date, event: [newEvent] });
       }
-      return dateArray;
-    };
+    });
+    setData(newState);
+    toast.success("You've successfully added a new event", {
+      position: toast.POSITION.TOP_CENTER,
+    });
+  };
+
+  //decline handler for overlapping events
+  const declineReplaceHandler = () => {
+    toggleModal();
+  };
 
+  const formSubmitHandler = async (e: React.FormEvent) => {
+    e.preventDefault();
+    let dates = [selectedDate];
```

Comment:
spread?

Reply:
I'm setting default value for dates to be [selectedDate] and after that I'm checking if recurring is selected. If it is then dates is reassigned to an array of all the dates that are going the event will repeat. This initial assigning is done so that the logic that updates the state stays the same no matter whether you have  selected recurring or not.